tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_remainder.h

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "noc_nonblocking_api.h"
+#include "ckernel_sfpu_converter.h"
+#include "ckernel_sfpu_recip.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_remainder(const uint value) {
+
+    // SFPU microcode
+    Converter c_value;
+    c_value.u = value;
+    vFloat s = c_value.f;
+    vFloat value_tmp = s;
+    s = sfpi::abs(s);
+
+    #pragma GCC unroll 0
+    for (int d = 0; d < ITERATIONS; d++) {
+        vFloat val = dst_reg[0];
+        vFloat v = sfpi::abs(val);
+
+        vFloat recip = sfpu_reciprocal<APPROXIMATION_MODE ? 2 : 3,true>(s);
```

Comment:
Is `value` just a scalar uint parameter? And not a VUInt (that contains 4 rows of dest data)? Because if that's the case, then I am unsure why this reciprocal is needed: `vFloat recip = sfpu_reciprocal<APPROXIMATION_MODE ? 2 : 3,true>(s);` to be done on device, when you can do `1/float(value)` on host, and then pass that value as a parameter to this function?

Reply:
Okay we will check with this method